src/operator/api/v2alpha1/clientintents_types.go

```diff
@@ -0,0 +1,888 @@
+}
+
+type HTTPTarget struct {
+	Path    string       `json:"path"`
+	Methods []HTTPMethod `json:"methods" yaml:"methods"`
+}
+
+type KafkaTarget struct {
+	Name string `json:"name" yaml:"name"`
+	//+optional
+	Topics []KafkaTopic `json:"topics,omitempty" yaml:"topics,omitempty"`
+}
+
+type KafkaTopic struct {
+	Name       string           `json:"name" yaml:"name"`
+	Operations []KafkaOperation `json:"operations" yaml:"operations"`
+}
+
+type ResolvedIPs struct {
+	DNS string   `json:"dns,omitempty" yaml:"dns,omitempty"`
+	IPs []string `json:"ips,omitempty" yaml:"ips,omitempty"`
+}
+
+type AzureKeyVaultPolicy struct {
+	//+optional
+	CertificatePermissions []AzureKeyVaultCertificatePermission `json:"certificatePermissions,omitempty" yaml:"certificatePermissions,omitempty"`
+	//+optional
+	KeyPermissions []AzureKeyVaultKeyPermission `json:"keyPermissions,omitempty" yaml:"keyPermissions,omitempty"`
+	//+optional
+	SecretPermissions []AzureKeyVaultSecretPermission `json:"secretPermissions,omitempty" yaml:"secretPermissions,omitempty"`
+	//+optional
+	StoragePermissions []AzureKeyVaultStoragePermission `json:"storagePermissions,omitempty" yaml:"storagePermissions,omitempty"`
+}
+
+// IntentsStatus defines the observed state of ClientIntents
+type IntentsStatus struct {
+	// upToDate field reflects whether the client intents have successfully been applied
+	// to the cluster to the state specified
+	// +optional
+	UpToDate bool `json:"upToDate"`
+	// The last generation of the intents that was successfully reconciled.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration"`
+	// +optional
+	ResolvedIPs []ResolvedIPs `json:"resolvedIPs,omitempty" yaml:"resolvedIPs,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:storageversion
+
+// ClientIntents is the Schema for the intents API
+type ClientIntents struct {
+	metav1.TypeMeta   `json:",inline" yaml:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+
+	Spec   *IntentsSpec  `json:"spec,omitempty" yaml:"spec,omitempty"`
+	Status IntentsStatus `json:"status,omitempty" yaml:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// ClientIntentsList contains a list of ClientIntents
+type ClientIntentsList struct {
+	metav1.TypeMeta `json:",inline" yaml:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+	Items           []ClientIntents `json:"items" yaml:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&ClientIntents{}, &ClientIntentsList{})
+}
+
+func (in *ClientIntents) GetWorkloadName() string {
+	return in.Spec.Workload.Name
+}
+
+func (in *ClientIntents) GetCallsList() []Target {
+	return in.Spec.Targets
+}
+
+func (in *ClientIntents) GetFilteredCallsList(intentTypes ...IntentType) []Target {
+	return lo.Filter(in.GetCallsList(), func(item Target, index int) bool {
+		for _, intentType := range intentTypes {
+			if intentType == IntentTypeHTTP {
```

Comment:
It doesn't translate to the same logic, because we iterate over the intent types and want to return only if there's a match (return true or continue). In this case, I have to use an extra `if` inside every case (for the nil check). I think that it is clearer when using `if` statements 🤔

Reply:
it's not exactly the case, you're not looking for "return true or continue" because each case has a unique intent type. So switching this to switch by type and for each type return `item.X != nil` will have the same meaning. 

That said, leaving this discussion as I think it's a matter of taste and not crucial in any way. Fell free to resolve this comment and move on.